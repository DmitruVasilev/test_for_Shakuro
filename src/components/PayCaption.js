import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Caption = styled.h1`
  display: flex;
  align-items: center;
  margin-bottom: 20px;
`;

const OperatorTitle = styled.span`
  margin: 0;
  font-weight: normal;
  font-size: 30px;
  line-height: 1;
`;

const PayCaption = ({ icon_url, title }) => (
  <Caption>
    <img src={icon_url} alt={title} />
    <OperatorTitle>{title}</OperatorTitle>
  </Caption>
);

PayCaption.propTypes = {
  icon_url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

export default PayCaption;
