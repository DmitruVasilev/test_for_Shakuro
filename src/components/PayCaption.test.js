/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import PayCaption from "./PayCaption";

const operatorsMock = {
  title: "MTS",
  icon_url: "https://citycard.ru/i/landing-features/operator-mts.svg"
};

describe("<OperatorList />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<PayCaption {...operatorsMock} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<PayCaption {...operatorsMock} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
