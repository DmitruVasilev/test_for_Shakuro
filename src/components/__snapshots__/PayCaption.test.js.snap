// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OperatorList /> renders correctly 1`] = `
<h1
  className="sc-bdVaJa lmOnTH"
>
  <img
    alt="MTS"
    src="https://citycard.ru/i/landing-features/operator-mts.svg"
  />
  <span
    className="sc-bwzfXH cgXZhI"
  >
    MTS
  </span>
</h1>
`;
